app/controllers/application_controller.rb

```diff
@@ -28,6 +28,22 @@ def alternate_locales
     I18n.available_locales - Array(I18n.locale)
   end
 
+  def alternate_url(new_locale, options = {}) # rubocop:disable Metrics/MethodLength
+    new_params = params.delete_if { |k| %w(utf8 authenticity_token).include?(k) }
```

Comment:
Not sure if these are `ActionController::Params` but could we use `Hash#except` here instead of the `delete_if` stuff?